openedx/adg/lms/applications/admin.py

```diff
@@ -68,7 +105,449 @@ class BusinessLineAdmin(admin.ModelAdmin):
+        return obj.user.profile.name
+
+    def date_received(self, obj):
+        return obj.user.application_hub.submission_date.strftime('%m/%d/%Y')
+    date_received.short_description = _('Date Received (MM/DD/YYYY)')
+
+    def changelist_view(self, request, extra_context=None):
+        """
+        Extend change list view of application listing page for ADG admin.
+
+        Extension is done to customize heading of the application listing page, depending on the application status
+        filter that the admin has selected.
+        """
+        application_status_map = {
+            UserApplication.OPEN: _('OPEN APPLICATIONS'),
+            UserApplication.ACCEPTED: _('ACCEPTED APPLICATIONS'),
+            UserApplication.WAITLIST: _('WAITLISTED APPLICATIONS')
+        }
+
+        extra_context = {'title': _('APPLICATIONS')}
+        if 'status__exact' in request.GET:
+            extra_context['title'] = application_status_map[request.GET['status__exact']]
+
+        return super(UserApplicationADGAdmin, self).changelist_view(request, extra_context=extra_context)
+
+    def changeform_view(self, request, object_id=None, form_url='', extra_context=None):
+        """
+        Extend change form view of application review page for ADG admin.
+
+        Extension is required to handle conditionally storing review information and to provide extra context to the
+        review form.
+
+        Arguments:
+            request(WSGIRequest): Http Request
+            object_id (str): ID of application under review
+            form_url (str): URL of the application review form
+            extra_context (NoneType): Extra context to be passed to the application review form template
+
+        Returns:
+            TemplateResponse: Template response to render application review form
+        """
+        if request.method == 'POST' and 'status' in request.POST:
+            self._save_application_review_info(object_id, request)
+
+        application = self._get_user_application(object_id)
+        extra_context = get_extra_context_for_application_review_page(application)
+
+        return super(UserApplicationADGAdmin, self).changeform_view(
+            request, object_id, form_url, extra_context=extra_context
+        )
+
+    def _save_application_review_info(self, application_id, request):
+        """
+        Save application review information.
+
+        Save new status of application (waitlist/accepted) and optional internal note.
+
+        Arguments:
+            application_id (str): ID of application under review
+            request (WSGIRequest): Post request containing application review information
+        """
+        application = self._get_user_application(application_id)
+
+        new_status = request.POST.get('status')
+        application.status = new_status
+
+        if 'internal_note' in request.POST:
+            note = request.POST.get('internal_note')
+            application.internal_admin_note = note
```

Comment:
Just a suggestion:
note = request.POST.get('internal_note')
if note:
            application.internal_admin_note = note

Reply:
I think if the note is not entered, the key 'internal_note' will not be a part of request.POST, and it will result in a key error. But I will test out what you suggested and keep in mind for the follow-up PR that I will be making for the unit tests.